source/core_mqtt.c

```diff
@@ -941,16 +941,15 @@ static MQTTStatus_t sendPublishAcks( MQTTContext_t * pContext,
 
             if( status != MQTTSuccess )
             {
-                LogError( ( "Failed to update state of publish %u.", packetId ) );
+                LogError( ( "Failed to update state of publish %u.",
+                            ( unsigned int ) packetId ) );
```

Comment:
Since this is originally a `uint16_t`, should we use unsigned short?
```suggestion
                LogError( ( "Failed to update state of publish %hu.",
                            ( unsigned short ) packetId ) );
```

Reply:
I think the definition of `short` is really dependent on the compiler, so I wanted to err on the side of caution. Although, we use `short` in other places for 16-byte data types, so I have updated it.

Reply:
An `unsigned short` is required to be able to hold values up to 65,535, according to [5.2.4.2](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf)